csharp/ql/src/Likely Bugs/ThreadUnsafeICryptoTransform.qhelp

```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>Classes that implement <code>System.Security.Cryptography.ICryptoTransform</code> are not thread safe.</p>
```

Comment:
Unfortunately the documentation in Microsoft docs is no longer specific about the instance members not being thread-safe.
The problem is because of the way the class is implemented using CAPI/CNG patterns. In an example of a hash class implementing this interface, typically there would be an instance specific hash object (say using BCryptCreateHash function), which can be called multiple times to add data to the hash (i.e. BCryptHashData), and finally calling the function that would finish the hash & get back the data (i.e. BCryptFinishHash). 
This implementation would potentially allow the same hash object to be called with data from multiple threads before calling the finish function, thus leading to potentially incorrect results.
For example, on a test I have multiple threads hashing "abc" on a static hash function, at least in one occasion I got the result for "abcabc".
It is very unlikely somebody outside Microsoft would write a class that implements ICryptoTransform, and even if they do, it is likely that they will follow the same CAPI/CNG pattern.

Reply:
I forgot to ask. Do you want me to add this explanation (after a cleanup) on the .qhelp file?

Reply:
That would indeed be a nice addition to the qhelp, thanks.

Reply:
Fixed. I expanded the documentation. Please feel free to correct (or let me know if I should correct) any typo or grammatical error.